OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -209,6 +209,9 @@ public bool HasAnyValidWeapons(Target t, bool checkForCenterTargetingWeapons = f
 			// PERF: Avoid LINQ.
 			foreach (var armament in Armaments)
 			{
+				if (armament.IsTraitDisabled)
+					continue;
```

Comment:
Not tested yet, but isn't this should be `!armament.IsTraitDisabled` or `return false;`. We want it to work if armament is enabled, not disabled.